### PR TITLE
front: migrate import train schedule files to ts

### LINKED ISSUE #4855: front: migrate ImportTrainSchedule files to ts
Migrate:
- [x] consts
- [x] generatePathfindingPayload
- [x] generateTrainSchedulesPayload
- [x] ImportTrainScheduleHelpers
- [ ] ImportTrainScheduleModal
- [x] ImportTrainScheduleModalFooter
- [x] ImportTrainScheduleStationSelector

### PR DESCRIPTION
**Pull request by clarani**
closes #4855

### CODE REVIEW

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleStationSelector.tsx [hunk 1]
```diff
@@ -1,43 +1,44 @@
...
-import { GRAOU_URL } from 'common/api/graouApi';
+import StationCard, { ImportStation } from 'common/StationCard';
+import { searchGraouStations } from 'common/api/graouApi';
 
-function ImportTrainScheduleStationSelector(props) {
-  const { id, onSelect, term, setTerm } = props;
+interface ImportTrainScheduleStationSelectorProps {
+  id: string;
+  term?: string;
+  onSelect: (stationName: ImportStation | undefined) => void;
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleStationSelector.tsx:14]:
```suggestion
  onSelect: (stationName?: ImportStation) => void;
```

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx [hunk 2]
```diff
@@ -40,17 +38,25 @@ export default function ImportTrainScheduleConfig({
 }: ImportTrainScheduleConfigProps) {
   const { t } = useTranslation(['operationalStudies/importTrainSchedule']);
   const [postSearch] = osrdEditoastApi.usePostSearchMutation();
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx:40]: Should you use this format for harmonization with the rest of the app ?
```suggestion
  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
```

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/generatePathfindingPayload.ts [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+  rollingStocks: LightRollingStock[],
+  defaultRollingStockId: number
 ) => {
-  const trainFromPathRef = trainsWithPathRef.find(
-    (train) => train.trainNumber === path.trainNumber
-  );
+  const trainFromPathRef = trainsWithPathRef.find((train) => train.trainNumber === trainNumber);
+
+  if (!trainFromPathRef) {
+    throw new Error('Train not found in trainsWithPathRef (getTrainAndRollingStockFromPath');
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/generatePathfindingPayload.ts:17]:
```suggestion
    throw new Error('Train not found in trainsWithPathRef (getTrainAndRollingStockFromPath)');
```

File: front/src/applications/operationalStudies/components/ImportTrainSchedule/generateTrainSchedulesPayload.ts [hunk 4]
```diff
@@ -1,12 +1,13 @@
+import { Step, TrainScheduleWithPath } from 'applications/operationalStudies/types';
+import { PathStep } from 'common/api/osrdEditoastApi';
 import { time2sec } from 'utils/timeManipulation';
 
 // Hope for indexes are the same !
 // Synchronisation is done with indexes between pathfinding not suggered positions, and required steps from importation
```
Comment (SharglutDev) [inline front/src/applications/operationalStudies/components/ImportTrainSchedule/generateTrainSchedulesPayload.ts:6]: "not suggested" ?

**Review by SharglutDev — COMMENTED** [review]
Good job, just a few comments :)

**Comment by nicolaswurtz** [comment]
#4885 merged instead

**Pull request closed without merging**